prov/efa/src/rdm/efa_rdm_pke.c

```diff
@@ -392,70 +386,65 @@ ssize_t efa_rdm_pke_sendv(struct efa_rdm_ep *ep,
 	conn = efa_av_addr_to_conn(ep->base_ep.av, pkt_entry_vec[0]->addr);
 	assert(conn && conn->ep_addr);
 
-	prev_send_wr = NULL;
-	for (i = 0; i < pkt_entry_cnt; ++i) {
-		pkt_entry = pkt_entry_vec[i];
+	qp = ep->base_ep.qp;
+	ibv_wr_start(qp->ibv_qp_ex);
+	for (pkt_idx = 0; pkt_idx < pkt_entry_cnt; ++pkt_idx) {
+		pkt_entry = pkt_entry_vec[pkt_idx];
 		assert(pkt_entry->pkt_size);
 		assert(efa_rdm_ep_get_peer(ep, pkt_entry->addr) == peer);
 
 		send = pkt_entry->send;
 		/* EFA device supports a maximum of 2 iov/SGE */
 		assert(send && send->iov_count <= 2);
-		if (send->iov_count == 0) {
+		if (send->iov_count < 2) {
 			send->iov_count = 1;
-			send->iov[0].iov_base = pkt_entry->wiredata;
-			send->iov[0].iov_len = pkt_entry->pkt_size;
-			send->desc[0] = pkt_entry->mr;
+			sg_list[0].addr = (uintptr_t)pkt_entry->wiredata;
+			sg_list[0].length = pkt_entry->pkt_size;
+			sg_list[0].lkey = ((struct efa_mr *)pkt_entry->mr)->ibv_mr->lkey;
+		} else {
+			for (iov_idx = 0; iov_idx < send->iov_count; iov_idx++) {
+				sg_list[iov_idx].addr = (uintptr_t)send->iov[iov_idx].iov_base;
+				sg_list[iov_idx].length = send->iov[iov_idx].iov_len;
+				sg_list[iov_idx].lkey = ((struct efa_mr *)send->desc[iov_idx])->ibv_mr->lkey;
+			}
 		}
 
+		qp->ibv_qp_ex->wr_id = (uintptr_t)pkt_entry;
+		ibv_wr_send(qp->ibv_qp_ex);
+		if (pkt_entry->pkt_size <= efa_rdm_ep_domain(ep)->device->efa_attr.inline_buf_size &&
+		    !efa_rdm_pke_has_hmem_mr(send)) {
+			for (iov_idx = 0; iov_idx < send->iov_count; ++iov_idx) {
+				inline_data_list[iov_idx].addr = (void *)sg_list[iov_idx].addr;
+				inline_data_list[iov_idx].length = sg_list[iov_idx].length;
+			}
+
+			ibv_wr_set_inline_data_list(qp->ibv_qp_ex, send->iov_count, inline_data_list);
```

Comment:
so this is equivalent to ibv_post_send with `IBV_SEND_INLINE` flag?

Reply:
Yes.